frontend/src/modules/Environments/components/EnvironmentRoleAddForm.js

```diff
@@ -12,53 +12,18 @@ import {
 import { Formik } from 'formik';
 import { useSnackbar } from 'notistack';
 import PropTypes from 'prop-types';
-import React, { useEffect, useState } from 'react';
+import React from 'react';
 import * as Yup from 'yup';
-import { Defaults } from 'design';
 import { SET_ERROR, useDispatch } from 'globalErrors';
-import { listEnvironmentGroups, useClient } from 'services';
+import { useClient } from 'services';
 import { addConsumptionRoleToEnvironment } from '../services';
+import { useFetchGroups } from '../../../utils/api';
```

Comment:
Should we change away from relative import here to keep consistent with how we set import statements, mainly:
- Edit `dataall/frontend/src/utils/index.js` to export * form `api.js`
- Edit import here as `import { useFetchGroups } from 'utils';`

Reply:
You are right, I also despise relative imports, IDE auto-generated it :D.

I am no JS expert but couldn't we create a mapping (in `package.json`) and then import like `@app/utils/api.js`? That way we can completely remove the `index.js` files from the project

Reply:
At the moment there are absolute imports across modules and relative imports inside each module. That's why you are seeing relative imports and sometimes we import from `../services` and sometimes form `services`

Reply:
I have never worked with JS mappings, but from what I am reading it still needs to specify each mapping in the json right? So it means replacing the `index.js` files in each package by entries in the json right? Maybe we can ask @maryamkhidir and @itsmo-amzn

Reply:
I don't think so, we just have to provide the path to the top directory and then do full imports from there.

Reply:
Hi @petrkalos, we recently conducted modularization for the frontend: https://github.com/data-dot-all/dataall/issues/398ff, in which we structured our project in terms of meaningful standalone modules instead of the typical functional approach (components, hooks, context, utils, ..).

So, we defined our main modules for absolute imports here: https://github.com/data-dot-all/dataall/blob/main/frontend/jsconfig.json

And for importing components or services within the same module, we went with relative imports.

It seemed a little far-fetched to import same-module files using absolute path instead of relative imports.

If you have a strong opinion on this, you can follow the docs here: https://create-react-app.dev/docs/importing-a-component/#absolute-imports to enable absolute imports for the entire project.

Reply:
Certainly a new PR, created an issue #1041